src/server/scripts/Northrend/Ulduar/Ulduar/boss_algalon_the_observer.cpp

```diff
@@ -765,14 +765,18 @@ struct npc_living_constellation : public CreatureAI
         }
     }
 
-    void SpellHit(Unit* caster, SpellInfo const* spell) override
+    void SpellHit(WorldObject* caster, SpellInfo const* spellInfo) override
     {
-        if (spell->Id != SPELL_CONSTELLATION_PHASE_EFFECT || caster->GetTypeId() != TYPEID_UNIT)
+        Unit* unitCaster = caster->ToUnit();
```

Comment:
This changes logic - `caster->GetTypeId() != TYPEID_UNIT` excludes players, `ToUnit()` does not